src/v/test_utils/http_imposter.cc

```diff
@@ -129,7 +129,7 @@ void http_imposter_fixture::fail_request_if(
   http_imposter_fixture::request_predicate predicate,
   http_test_utils::response response) {
     _fail_requests_when.push_back(std::move(predicate));
-    _fail_responses[_fail_requests_when.size()] = std::move(response);
+    _fail_responses[_fail_requests_when.size() - 1] = std::move(response);
```

Comment:
this change is also part of the compacted topic reupload PR, it was a bug with the imposter.